File: valid-palindrome/valid-palindrome.java
class Solution {
    public boolean isPalindrome(String s) {
        char[] sArray = s.toLowerCase().toCharArray();
        
        int left = 0;
        int right = sArray.length - 1;
        
        while(left < right){
            Character first = sArray[left];
            Character second = sArray[right];
            if(!Character.isLetterOrDigit(first) || Character.isWhitespace(first)){
                left++;
                continue;
            }
            if(!Character.isLetterOrDigit(second)  || Character.isWhitespace(second)){
                right--;
                continue;
            }
            
            if(!first.equals(second)){
                return false;
            }
            
            left++;right--;
            
        }
        
        return true;
        
    }
}

File: valid-palindrome/valid-palindrome.kt
class Solution {
    fun isPalindrome(s: String): Boolean {
        
        val sArray = s.toLowerCase().toCharArray()
        
        var left = 0
        var right = sArray.size-1

        while(left < right){
           var first = sArray[left] 
           var second = sArray[right] 
           if(!first.isLetterOrDigit()){
               left++
               continue
           }  
           if(!second.isLetterOrDigit()){
               right--
               continue
           }  
           
           if(first != second) return false
            
           left++
           right--
        }
        

        return true;

        
    }
}